org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BundleUtils.java

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.jdt.ls.core.internal.handlers;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.MultiStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleException;
+
+/**
+ * BundleContext and Bundle utilities
+ */
+public class BundleUtils {
+
+	private static final String REFERENCE_PREFIX = "reference:";
+
+	/**
+	 * Load list of bundle
+	 *
+	 * @param bundleList
+	 *            The bundle list of location to load
+	 * @throws CoreException
+	 *             throw the <code>CoreException</code> if failed to load any
+	 *             bundles.
+	 */
+	public static void loadBundles(List<String> bundleList) throws CoreException {
+		if (bundleList == null || bundleList.size() == 0) {
+			return;
+		}
+
+		BundleContext context = JavaLanguageServerPlugin.getBundleContext();
+		MultiStatus status = new MultiStatus(context.getBundle().getSymbolicName(), IStatus.OK, "Load bundle list", null);
+		for (String bundlePath : bundleList) {
+			try {
+				String bundleLocation = getBundleLocation(bundlePath, true);
+				if (StringUtils.isEmpty(bundleLocation)) {

```

Comment:
bundleLocation will always at least contain "reference:", so the check is either useless or wrong.